lib/ansible/plugins/action/synchronize.py

```diff
@@ -326,7 +328,6 @@ def run(self, tmp=None, task_vars=None):
             private_key = self._play_context.private_key_file
 
             if private_key is not None:
-                private_key = os.path.expanduser(private_key)
```

Comment:
The fix only works if `expanduser` is not called from the action. In fact, from what I understand in that case, the module is sent to the delegate host and the action builds the parameters that will be used for the call in the machine running ansible.

So calling `expanduser` both from the action and the module will expand it on the local machine and then on the delegate. But because it was already expanded on the local machine, expanding it on the delegate won't have any effect and the path will be relative to the local machine (the one running ansible).

In my test, I checked that it works well with and without delegate, so I don't expect any regression with this patch. Also, you make a valid point and we could remove the `private_key` variable and directly do:

```
_tmp_args['private_key'] = self._play_context.private_key_file
```

Without even checking if the key is none as we already check that on the module. If you prefer this fix, I can update my PR.